### PR TITLE
[SPARK-18873][SQL][TEST] New test cases for scalar subquery (part 2 of 2) - scalar subquery in predicate context

### PR DESCRIPTION
**Pull request by nsyca**
## What changes were proposed in this pull request?
This PR adds new test cases for scalar subquery in predicate context

## How was this patch tested?
The test result is compared with the result run from another SQL engine (in this case is IBM DB2). If the result are equivalent, we assume the result is correct.

### CODE REVIEW

**Comment by nsyca** [comment]
Below are a modified version of the test cases to run on DB2 and the result from DB2, as a second source to compare to the result from Spark.
[Modified test file to run on DB2](https://github.com/apache/spark/files/751915/scalar-subquery-predicate.sql.db2.txt)
[Result from DB2](https://github.com/apache/spark/files/751918/scalar-subquery-predicate.sql.db2.out.txt)

File: sql/core/src/test/resources/sql-tests/inputs/scalar-subquery.sql [hunk 1]
```diff
@@ -1,20 +0,0 @@
...
-
--- SPARK-18814.2: Adding stack of aggregates
-SELECT pk, cv
-FROM   p, c
-WHERE  p.pk = c.ck
-AND    c.cv = (SELECT max(avg)
-	       FROM   (SELECT   c1.cv, avg(c1.cv) avg
-		       FROM     c c1
-		       WHERE    c1.ck = p.pk
-                       GROUP BY c1.cv));
```
Comment (nsyca) [inline sql/core/src/test/resources/sql-tests/inputs/scalar-subquery.sql]: Merged the test cases with the new test cases and placed under the directory "scalar-subquery".

**Comment by nsyca** [comment]
@dilipbiswal Could you please cross-check the results from both sources?
@gatorsmile, @hvanhovell Could you please review?

File: sql/core/src/test/resources/sql-tests/results/subquery/scalar-subquery/scalar-subquery-predicate.sql.out [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+GROUP BY t1a, t1c
+HAVING   max(t1b) <= (SELECT   max(t2b)
+                      FROM     t2
+                      WHERE    t2c = t1c
+                      GROUP BY t2c)
+-- !query 24 schema
+struct<t1a:string>
+-- !query 24 output
+val1b
+val1c
```
Comment (dilipbiswal) [inline sql/core/src/test/resources/sql-tests/results/subquery/scalar-subquery/scalar-subquery-predicate.sql.out]: I have compared the result set matched with the result from DB2.

**Comment by gatorsmile** [comment]
retest this please

File: sql/core/src/test/resources/sql-tests/inputs/subquery/scalar-subquery/scalar-subquery-predicate.sql [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+              GROUP BY t2c);
+
+-- TC 02.05
+SELECT t1a
+FROM   t1
+WHERE  t1b <= (SELECT   max(t2b)
+               FROM     t2
+               WHERE    t2c = t1c
+               GROUP BY t2c)
+AND    t1b >= (SELECT   min(t2b)
```
Comment (dongjoon-hyun) [inline sql/core/src/test/resources/sql-tests/inputs/subquery/scalar-subquery/scalar-subquery-predicate.sql]: Nit. I like this indentation.
For the other examples, `AND` seems to be aligned with `t1b` at line 190.

File: sql/core/src/test/resources/sql-tests/inputs/subquery/scalar-subquery/scalar-subquery-predicate.sql [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+-- TC 02.07
+-- set op
+SELECT t1a
+FROM   t1
+WHERE  t1a <= (SELECT   max(t2a)
+               FROM     t2
+               WHERE    t2c = t1c
+               GROUP BY t2c)
+UNION ALL
```
Comment (dongjoon-hyun) [inline sql/core/src/test/resources/sql-tests/inputs/subquery/scalar-subquery/scalar-subquery-predicate.sql]: Can we have `UNION` here too?

Comment (nsyca) [inline sql/core/src/test/resources/sql-tests/inputs/subquery/scalar-subquery/scalar-subquery-predicate.sql]: @dongjoon-hyun Thank you for your comment. I have added another test case using UNION DISTINCT.

I would appreciate if you could share your insight on what you think the UNION test case will process differently from the UNION ALL test case with respect to the testing of scalar subquery.

Note that the correctness of the result of UNION DISTINCT can be inferred from applying the "uniqueness" operator on the result of the existing UNION test case.

**Comment by hvanhovell** [comment]
LGTM - merging to master

**Pull request closed without merging**